### PR TITLE
RTL check alt_max before using

### PR DESCRIPTION
**Pull request by dagar**
Partial fix for #8257.

### CODE REVIEW

File: src/modules/navigator/rtl.cpp [hunk 1]
```diff
@@ -40,29 +40,18 @@
...
-#include <mathlib/mathlib.h>
-#include <systemlib/err.h>
-#include <systemlib/mavlink_log.h>
-#include <uORB/topics/home_position.h>
-#include <uORB/topics/vtol_vehicle_status.h>
-#include <uORB/uORB.h>
+#include <cfloat>
 
-static constexpr float DELAY_SIGMA = 0.01f;
+using math::max;
```
Comment (MaEtUgR) [inline src/modules/navigator/rtl.cpp]: probably you can't take away `#include <mathlib/mathlib.h>` and use the function defined in it...
or is it still in the include chain?

**Comment by dagar** [comment]
In a quick test the climb issue and multiple RTL in the same flight strangeness seem to be resolved, but there are many edge cases - https://github.com/PX4/Firmware/pull/8272

**Comment by Stifael** [comment]
it looks ok, but the get_rtl_function is not really needed anymore. 
The function [here](https://github.com/PX4/Firmware/pull/8272/files#diff-6dcf9c0efb29cd9722cf2d7c980be527R68) is a duplicate of [here](https://github.com/PX4/Firmware/blob/master/src/modules/navigator/mission_block.cpp#L699-L720), which is checked at the end of each navigation mode. That is why I removed that function [here](https://github.com/PX4/Firmware/pull/8101/files#diff-6dcf9c0efb29cd9722cf2d7c980be527L77). 

The point is that the altitude limit with -1 was introduced, but rtl pr was never merged, which resulted in a conflict.

**Comment by Stifael** [comment]
also to mention is that this [PR](https://github.com/PX4/Firmware/pull/8101) is a pr that comes from a fork that was tested more than half a year. My point is that this PR is good, but maybe we should combine this PR with https://github.com/PX4/Firmware/pull/8101.

**Comment by dagar** [comment]
Updated to remove get_rtl_altitude().
